src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp

```diff
@@ -4254,5 +4255,24 @@ GUI_TEST_CLASS_DEFINITION(test_7806) {
     CHECK_SET_ERR(size == 4, "chrM.fa in SAM dir is changed, size: " + QString::number(size));
 }
 
+GUI_TEST_CLASS_DEFINITION(test_7830) {
+    // 1. Open COI.aln
+    GTFileDialog::openFile(os, dataDir + "samples/CLUSTALW/", "COI.aln");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    // 2. Right button-> Statistics-> Generate distance matrix
+    // 3. Click "Generate"
+    GTUtilsDialog::add(os, new PopupChooser(os, { MSAE_MENU_STATISTICS, "Generate distance matrix" }, GTGlobals::UseMouse));
+    GTUtilsDialog::add(os, new DistanceMatrixDialogFiller(os, true, true, true));
+    GTMenu::showContextMenu(os, GTUtilsMdi::activeWindow(os));
+
+    // Expected: legend exists
+    QTextBrowser* v = GTUtilsMdi::activeWindow(os)->findChild<QTextBrowser*>();
```

Comment:
Never use findChild in UI tests: use GTWidget::find
findChild is unsafe because it iterates UI object structure in non-ui thread.

Reply:
OK

Reply:
This part has been removed